src/data/set/basic.lean

```diff
@@ -1549,6 +1549,12 @@ iff.not_left not_subsingleton_iff.symm
 alias not_nontrivial_iff ↔ _ subsingleton.not_nontrivial
 alias not_subsingleton_iff ↔ _ nontrivial.not_subsingleton
 
+protected lemma subsingleton_or_nontrivial (s : set α) : s.subsingleton ∨ s.nontrivial :=
```

Comment:
Do we already have this lemma for finset?

Reply:
No, and there's no point having it because this lemma already applies (`a ∈ ↑s ↔ a ∈ s`), as opposed to `eq_singleton_or_nontrivial` (`s = {a}` is not defeq to `↑s = {a}`).